SDSS test failures with python2 on appveyor

These test failures appeared a few weeks ago in totally unrelated PRs. Only affects the python2 build on appveyor while travis passes without any problems. 
For the time being I've allowed this build to fail, once this issue is close, it should be removed, too.

The first traceback:


```
______________________________ test_sdss_sql[1] _______________________________
patch_get = <_pytest.monkeypatch.monkeypatch instance at 0x000000000AC85688>
patch_get_readable_fileobj = <_pytest.monkeypatch.monkeypatch instance at 0x000000000AC85688>
dr = 1
    @pytest.mark.parametrize("dr", [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12])
    def test_sdss_sql(patch_get, patch_get_readable_fileobj, dr):
        query = """
                select top 10
                  z, ra, dec, bestObjID
                from
                  specObj
                where
                  class = 'galaxy'
                  and z > 0.3
                  and zWarning = 0
                """
        xid = sdss.SDSS.query_sql(query, data_release=dr)
        data = Table.read(data_path(DATA_FILES['images_id']),
                          format='ascii.csv', comment='#')
>       compare_xid_data(xid, data)
astroquery\sdss\tests\test_sdss.py:193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery\sdss\tests\test_sdss.py:144: in compare_xid_data
    assert_allclose(xid[col], data[col])
C:\conda\envs\test\lib\site-packages\numpy\testing\utils.py:667: in safe_comparison
    return comparison(*args, **kwargs)
C:\conda\envs\test\lib\site-packages\numpy\testing\utils.py:1387: in compare
    equal_nan=equal_nan)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
a = <Column name='objid' dtype='int64' length=18>
1237653651835781245
123765365183...1237649953859174621
1237649953859174622
1237649953859174623
1237649953859240023
b = <Column name='objid' dtype='str19' length=18>
1237653651835781245
123765365183...1237649953859174621
1237649953859174622
1237649953859174623
1237649953859240023
rtol = 1e-07, atol = 0, equal_nan = False
    def isclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
    ...

        # Make sure y is an inexact type to avoid bad behavior on abs(MIN_INT).
        # This will cause casting of x later. Also, make sure to allow subclasses
        # (e.g., for numpy.ma).
        dt = multiarray.result_type(y, 1.)
        y = array(y, dtype=dt, copy=False, subok=True)
    
        xfin = isfinite(x)
>       yfin = isfinite(y)
E       TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
C:\conda\envs\test\lib\site-packages\numpy\core\numeric.py:2451: TypeError
______________________________ test_sdss_sql[2] _______________________________

```
https://ci.appveyor.com/project/Astropy/astroquery/build/1.0.286/job/wftsjmgwu542iug4

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -22,13 +22,6 @@ environment:
         ASTROPY_VERSION: "stable"
         NUMPY_VERSION: "stable"
 
-matrix:
-  # Allow to fail the 2.7 build until #823 is fixed
-  allow_failures:
-      - PYTHON_VERSION: "2.7"
-        ASTROPY_VERSION: "stable"
-        NUMPY_VERSION: "stable"
-
 platform:
     -x64
 

--- a/astroquery/sdss/tests/test_sdss.py
+++ b/astroquery/sdss/tests/test_sdss.py
@@ -205,6 +205,10 @@ def test_sdss_sql(patch_get, patch_get_readable_fileobj, dr):
     xid = sdss.SDSS.query_sql(query, data_release=dr)
     data = Table.read(data_path(DATA_FILES['images_id']),
                       format='ascii.csv', comment='#')
+    # The following line is needed for systems where the default integer type
+    # is int32, the column will then be interpreted as string which makes the
+    # test fail.
+    data['objid'] = data['objid'].astype(np.int64)
     compare_xid_data(xid, data)
     url_tester(dr)
 
@@ -242,6 +246,11 @@ def test_sdss_specobj(patch_get, dr):
     xid = sdss.SDSS.query_specobj(plate=2340, data_release=dr)
     data = Table.read(data_path(DATA_FILES['spectra_id']),
                       format='ascii.csv', comment='#')
+    # The following line is needed for systems where the default integer type
+    # is int32, the column will then be interpreted as string which makes the
+    # test fail.
+    data['specobjid'] = data['specobjid'].astype(np.int64)
+    data['objid'] = data['objid'].astype(np.int64)
     compare_xid_data(xid, data)
     url_tester(dr)
 
@@ -252,6 +261,10 @@ def test_sdss_photoobj(patch_get, dr):
         run=1904, camcol=3, field=164, data_release=dr)
     data = Table.read(data_path(DATA_FILES['images_id']),
                       format='ascii.csv', comment='#')
+    # The following line is needed for systems where the default integer type
+    # is int32, the column will then be interpreted as string which makes the
+    # test fail.
+    data['objid'] = data['objid'].astype(np.int64)
     compare_xid_data(xid, data)
     url_tester(dr)
 
@@ -261,6 +274,10 @@ def test_list_coordinates(patch_get, dr):
     xid = sdss.SDSS.query_region(coords_list, data_release=dr)
     data = Table.read(data_path(DATA_FILES['images_id']),
                       format='ascii.csv', comment='#')
+    # The following line is needed for systems where the default integer type
+    # is int32, the column will then be interpreted as string which makes the
+    # test fail.
+    data['objid'] = data['objid'].astype(np.int64)
     compare_xid_data(xid, data)
 
 
@@ -269,6 +286,10 @@ def test_column_coordinates(patch_get, dr):
     xid = sdss.SDSS.query_region(coords_column, data_release=dr)
     data = Table.read(data_path(DATA_FILES['images_id']),
                       format='ascii.csv', comment='#')
+    # The following line is needed for systems where the default integer type
+    # is int32, the column will then be interpreted as string which makes the
+    # test fail.
+    data['objid'] = data['objid'].astype(np.int64)
     compare_xid_data(xid, data)
     url_tester(dr)
 
@@ -293,6 +314,10 @@ def test_query_crossid(patch_post, dr):
     xid = sdss.SDSS.query_crossid(coords_column, data_release=dr)
     data = Table.read(data_path(DATA_FILES['images_id']),
                       format='ascii.csv', comment='#')
+    # The following line is needed for systems where the default integer type
+    # is int32, the column will then be interpreted as string which makes the
+    # test fail.
+    data['objid'] = data['objid'].astype(np.int64)
     compare_xid_data(xid, data)
     url_tester_crossid(dr)
 

